### PR TITLE
Allow Active Tab of a Custom Window To Be Changed Through Script

### PR DESCRIPTION
**Pull request by jazzysoggy**
#17868

Enhancement: Active Tab of a Custom Window can now be changed through a new function named ``tabIndex_set``

### CODE REVIEW

File: src/openrct2-ui/scripting/ScWindow.hpp [hunk 1]
```diff
@@ -277,6 +277,14 @@ namespace OpenRCT2::Scripting
             }
             return 0;
         }
+        void tabIndex_set(size_t tab) const
```
Comment (IntelOrca) [inline src/openrct2-ui/scripting/ScWindow.hpp:280]:
```suggestion
        void tabIndex_set(int32_t tab) const
```

**Comment by Basssiiie** [comment]
You'll need to register the setter [here](https://github.com/OpenRCT2/OpenRCT2/blob/344cb6e25ada5656143d9051a04f081370978c75/src/openrct2-ui/scripting/ScWindow.hpp#L339) as well, otherwise it won't work. You have tested it, right?

Oh and the plugin API version needs to be incremented as well. :)

**Review by IntelOrca — APPROVED** [review]

**Review by IntelOrca — CHANGES_REQUESTED** [review]
As @Basssiiie pointed out, you need to register the setter

**Comment by jazzysoggy** [comment]
@Basssiiie It's not tested since I don't really know how to use the Plugin API. So I'm not sure if it works the way it should

**Comment by Sadret** [comment]
Seems to work fine, but the game crashes when an invalid tab index is provided. Make sure to validate the input.

**Comment by jazzysoggy** [comment]
Added a validation check for tab index

**Review by IntelOrca — APPROVED** [review]

File: distribution/changelog.txt [hunk 2]
```diff
@@ -10,6 +10,7 @@
 - Improved: [#15358] Park and scenario names can now contain up to 128 characters.
 - Improved: [#16840] Add support for rectangular heightmaps.
 - Improved: [#17575] You can now search for Authors in Object Selection.
+- Improved: [#17868] [Plugin] You can now change active tab of a custom window programmatically
```
Comment (Gymnasiast) [inline distribution/changelog.txt:13]:
```suggestion
- Improved: [#17868] [Plugin] You can now change active tab of a custom window programmatically.
```

**Comment by Sadret** [comment]
I'm on current develop (97bc2b5) and this does not seem to work.

**Comment by Gymnasiast** [comment]
@Sadret Please create an issue for that.

**Pull request merged**